### PR TITLE
New England Foods

### LINKED ISSUE #45716: Missing New England foods
#45680 has me thinking about common New England foods that are missing from CDDA.

Here's a list I came up with off the top of my head, feel free to add to it:

- New York System weiners (and associated diners - should be somewhat rare)
- crab cakes (small stands / food trucks? Especially in amusement park-type locations)
- oyster crackers should be common in kitchens, especially near clam chowder
- Boston Baked Beans candy in gas stations and/or vending machines
- American Chop Suey dish crafted by player out of pasta, unprepared tomatoes (raw/canned/etc) and sundry spices and crunchy vegetables (celery, peppers, etc.)
- Hoodsie Cup ice cream in ice cream trucks and gas stations
- quahogs and stuffed quahogs ("stuffies")
- rhubarb plants and rhubarb pie recipe, also sweet rhubarb from dipping stalk directly in sugar before eating
- frozen lemonade
- Johnny cakes (pan-fried cornmeal biscuits)
- lobstah

### PR DESCRIPTION
**Pull request by hjk321**
#### Summary

SUMMARY: Content "Added regional foods from the New England area"

#### Purpose of change

Closes #45716.
Since CDDA is set in the New England area, several foods popular in the area were added, including itemgroup entries and recipes where applicable.

#### Describe the solution

Adds:
- New York Wiener (a hot dog with ground meat and chopped onions)
- Stuffed clams (Similar to fried fish)
- Crab cakes (No recipe as of right now since there's no crab)
- Candy-coated peanuts (based on Boston baked beans candy)
- Oyster crackers
- Frozen lemonade (essentially slushie except 10 times better)
- Sundae Cups (small 3oz dairy dessert cups based on hoodsie cups)

Foods were added to various itemgroups such as pantries, diners, barbecues, vending machines, etc.
The only items with recipes are the new york wiener, stuffed clams, and frozen lemonade.

#### Testing

Rudimentary tests of the foods and spawns seem to work okay, probably because these food items all copy-from pre-existing food items. However, this is my first major PR, and I only tested it for about 15 minutes so it's possible that I missed something.

#### Additional context

See  #45716.

### CODE REVIEW

File: data/json/recipes/food/frozen.json [hunk 1]
```diff
@@ -128,6 +128,25 @@
...
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_OTHER",
+    "skill_used": "cooking",
+    "difficulty": 1,
+    "time": "10 m",
+    "autolearn": true,
+    "qualities": [ { "id": "COOK", "level": 3 } ],
+    "tools": [ [ [ "rock_quern", -1 ], [ "clay_quern", -1 ], [ "mortar_pestle", -1 ], [ "food_processor", 10 ] ] ],
+    "components": [
+      [ [ "milk_standard", 1, "LIST" ], [ "con_milk", 1 ], [ "can_coconut", 1 ] ],
```
Comment (emptytriangle) [inline data/json/recipes/food/frozen.json:144]: looks like you left the milk in the recipe, I'd remove the milk and add a requirement for one real lemon or irradiated lemon per portion in addition to the lemonade

**Review by emptytriangle — COMMENTED** [review]
one comment, the rest looks good to me

File: data/json/recipes/food/frozen.json (hunk 1 above)
Comment (hjk321) [reply data/json/recipes/food/frozen.json]: Oops! I was thinking of the chik-fil-a frozen lemonade (which has dairy) and never bothered to look up the real thing! Fixed.

**Review by emptytriangle — APPROVED** [review]

**Comment by Positronic-Girl** [comment]
Adding crabs should be simple enough for this PR, no?

**Comment by Xaleth** [comment]
> Adding crabs should be simple enough for this PR, no?

why?

**Comment by emptytriangle** [comment]
> Adding crabs should be simple enough for this PR, no?
you should be able to catch crabs and there should be salt water somewhere, I mean yeah you could just add some canned crab or something but crabs probably deserve their own pr

**Comment by hjk321** [comment]
Honestly, adding entire new creatures along with AI, cooked versions, all that jazz, is outside the scope of what I'm interested in contributing right now, since I'm working on other stuff in C++ at the moment. @Mom-Bun, if you wanna clone my branch and open a PR on my fork, we can go from there. Otherwise, it's probably okay keeping it worldgen only for now.

File: data/json/items/comestibles/junkfood.json [hunk 2]
```diff
@@ -1135,6 +1135,20 @@
...
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "hotdogs_newyork",
+    "name": { "str": "new york wiener" },
+    "copy-from": "hotdogs_cooked",
+    "weight": "568 g",
+    "looks-like": "hotdogs_cooked",
+    "calories": 721,
+    "description": "What's better than a hot dog? A hot dog smothered in ground meat and chopped onions, of course! Tastes better than a normal hot dog.",
```
Comment (BrettDong) [inline data/json/items/comestibles/junkfood.json:1146]: Needs two spaces between sentences to pass CI check
```suggestion
    "description": "What's better than a hot dog?  A hot dog smothered in ground meat and chopped onions, of course!  Tastes better than a normal hot dog.",
```

File: data/json/items/comestibles/junkfood.json [hunk 3]
```diff
@@ -1135,6 +1135,20 @@
     "fun": 12,
     "vitamins": [ [ "vitC", 85 ], [ "calcium", 22 ], [ "iron", 15 ] ]
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "hotdogs_newyork",
+    "name": { "str": "new york wiener" },
+    "copy-from": "hotdogs_cooked",
+    "weight": "568 g",
+    "looks-like": "hotdogs_cooked",
```
Comment (BrettDong) [inline data/json/items/comestibles/junkfood.json:1144]: It's `looks_like`, underscore.
```suggestion
    "looks_like": "hotdogs_cooked",
```

File: data/json/items/comestibles/junkfood.json (hunk 2 above)
Comment (BrettDong) [inline data/json/items/comestibles/junkfood.json:1146]: Needs two spaces between sentences to pass CI check.
```suggestion
    "description": "What's better than a hot dog?  A hot dog smothered in ground meat and chopped onions, of course!  Tastes better than a normal hot dog.",
```

File: data/json/items/comestibles/meat_dishes.json [hunk 4]
```diff
@@ -347,6 +347,33 @@
...
+    "type": "COMESTIBLE",
+    "id": "crab_cakes",
+    "name": { "str": "crab cake" },
+    "copy-from": "fish_fried",
+    "weight": "150 g",
+    "calories": 572,
+    "description": "A delicious deep-fried biscuit loaded with crab meat.",
+    "//": "Currently only found in worldgen and is uncraftable. If crab gets added please make a recipe for crab cakes.",
+    "volume": "175 ml",
+    "looks-like": "fish_fried",
```
Comment (BrettDong) [inline data/json/items/comestibles/meat_dishes.json:360]:
```suggestion
    "looks_like": "fish_fried",
```

File: data/json/items/comestibles/meat_dishes.json [hunk 5]
```diff
@@ -347,6 +347,33 @@
...
+  {
+    "type": "COMESTIBLE",
+    "id": "stuffed_clams",
+    "name": "stuffed clam",
+    "copy-from": "fish_fried",
+    "weight": "150 g",
+    "calories": 572,
+    "description": "Quahog clams stuffed with bread crumbs and fried.",
+    "volume": "175 ml",
+    "looks-like": "fish_fried",
```
Comment (BrettDong) [inline data/json/items/comestibles/meat_dishes.json:372]:
```suggestion
    "looks_like": "fish_fried",
```

File: data/json/items/comestibles/wheat.json [hunk 6]
```diff
@@ -370,6 +370,20 @@
...
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "oyster_crackers",
+    "name": { "ctxt": "oyster crackers", "str_sp": "oyster crackers" },
+    "copy-from": "crackers",
+    "weight": "25 g",
+    "container": "plastic_bag_vac",
+    "description": "These bite-sized crackers go well with soups, but will make you thirsty on their own.",
+    "looks-like": "crackers",
```
Comment (BrettDong) [inline data/json/items/comestibles/wheat.json:381]:
```suggestion
    "looks_like": "crackers",
```

**Review by BrettDong — CHANGES_REQUESTED** [review]
Need to fix CI check errors

File: data/json/items/comestibles/junkfood.json [hunk 7]
```diff
@@ -1158,5 +1172,29 @@
...
+    "weight": "30 g",
+    "color": "brown",
+    "spoils_in": "240 days",
+    "container": "box_small",
+    "healthy": -2,
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "quench": -4,
+    "calories": 220,
+    "description": "Peanuts coated in a sweet, crunchy candy coating. A regional delicacy.",
```
Comment (BrettDong) [inline data/json/items/comestibles/junkfood.json:1189]:
```suggestion
    "description": "Peanuts coated in a sweet, crunchy candy coating.  A regional delicacy.",
```

**Review by Xaleth — APPROVED** [review]

**Review by BrettDong — APPROVED** [review]

**Pull request merged**